botocore/docs/client.py

```diff
@@ -0,0 +1,102 @@
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import inspect
+
+from botocore.docs.utils import get_official_service_name
+from botocore.docs.method import document_custom_signature
+from botocore.docs.method import document_model_driven_method
+from botocore.docs.method import get_instance_public_methods
+
+
+class ClientDocumenter(object):
+    def __init__(self, client):
+        self._client = client
+        self._service_name = self._client.meta.service_model.service_name
+
+    def document_client(self, section):
+        """Documents a client and its methods
+
+        :param section: The section to write to.
+        """
+        self._add_title(section)
+        self._add_class_signature(section)
+        client_methods = get_instance_public_methods(self._client)
+        self._add_client_intro(section, client_methods)
+        self._add_client_methods(section, client_methods)
+
+    def _add_title(self, section):
+        section.style.h2('Client')
+
+    def _add_client_intro(self, section, client_methods):
+        section = section.add_new_section('intro')
+        # Write out the top level description for the client.
+        official_service_name = get_official_service_name(
+            self._client.meta.service_model)
+        section.write(
+            'A low-level client representing %s' % official_service_name)
+
+        # Write out the client example instantiation.
+        section.style.start_codeblock()
+        section.style.new_line()
+        section.write(
+            'client = session.create_client(\'{service}\')'.format(
+                service=self._service_name)
+        )
+        section.style.end_codeblock()
+
+        # List out all of the possible client methods.
+        section.style.new_line()
+        section.write('These are the available methods:')
+        section.style.new_line()
+        for method_name in sorted(client_methods):
+            section.style.li(':py:meth:`%s.Client.%s`' % (
+                self._service_name, method_name))
+
+    def _add_class_signature(self, section):
+        section.style.start_sphinx_py_class(
+            class_name='%s.Client' % self._service_name)
+
+    def _add_client_methods(self, section, client_methods):
+        section = section.add_new_section('methods')
+        for method_name in sorted(client_methods):
+            self._add_client_method(
+                section, method_name, client_methods[method_name])
+
+    def _add_client_method(self, section, method_name, method):
+        section = section.add_new_section(method_name)
+        if self._is_custom_method(method_name):
+            self._add_custom_method(section, method_name, method)
+        else:
+            self._add_model_driven_method(section, method_name)
+
+    def _is_custom_method(self, method_name):
+        return method_name not in self._client._PY_TO_OP_NAME
```

Comment:
`_PY_TO_OP_NAME` is used in a few places in this change.  Let's just expose a public API for this.  Perhaps we can add something to `client.meta` that allows you to get this mapping?